### PR TITLE
chore: rename `basic-workflow-tests.sh` to `e2e-custom.sh`

### LINKED ISSUE #2430: Rename `basic-workflow-tests.sh` to `e2e-custom.sh`
As we further refine our CI ecosystem, we want to rename this legacy script to reflect it's planned usage - as a tool for folks to use locally as well as the "custom" E2E job.

As part of this work, we will need to make updates in the following libraries as well:
- [x] SDG: https://github.com/instructlab/sdg/pull/306
- [x] Training: https://github.com/instructlab/training/pull/270
- [x] Eval: https://github.com/instructlab/eval/pull/152

### PR DESCRIPTION
**Pull request by nathan-weinberg**
**Issue resolved by this Pull Request:**
Resolves #2430

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [x] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
The small CI job will fail here due to the script renaming - once I have the usual two approvals I will merge manually (assuming all other checks pass)

**Review by cdoern — APPROVED** [review]
nice, thanks

**Review by ktam3 — APPROVED** [review]

**Comment by nathan-weinberg** [comment]
@Mergifyio backport release-v0.19

**Pull request merged**
